frame/assets/src/lib.rs

```diff
@@ -844,6 +953,76 @@ decl_module! {
 				Ok(())
 			})
 		}
+
+		/// Set the metadata for an asset.
+		///
+		/// NOTE: There is no `unset_metadata` call. Simply pass an empty name, symbol,
+		/// and 0 decimals to this function to remove the metadata of an asset and
+		/// return your deposit.
+		///
+		/// Origin must be Signed and the sender should be the Owner of the asset `id`.
+		///
+		/// Funds of sender are reserved according to the formula:
+		/// `MetadataDepositBase + MetadataDepositPerByte * (name.len + symbol.len)` taking into
+		/// account any already reserved funds.
+		///
+		/// - `id`: The identifier of the asset to update.
+		/// - `name`: The user friendly name of this asset. Limited in length by `StringLimit`.
+		/// - `symbol`: The exchange symbol for this asset. Limited in length by `StringLimit`.
+		/// - `decimals`: The number of decimals this asset uses to represent one unit.
+		///
+		/// Emits `MaxZombiesChanged`.
+		///
+		/// Weight: `O(1)`
+		#[weight = T::WeightInfo::set_metadata(name.len() as u32, symbol.len() as u32)]
+		fn set_metadata(origin,
+			#[compact] id: T::AssetId,
+			name: Vec<u8>,
+			symbol: Vec<u8>,
+			decimals: u8,
+		) -> DispatchResult {
+			let origin = ensure_signed(origin)?;
+
+			ensure!(name.len() <= T::StringLimit::get(), Error::<T>::BadMetadata);
+			ensure!(symbol.len() <= T::StringLimit::get(), Error::<T>::BadMetadata);
+
+			let d = Asset::<T>::get(id).ok_or(Error::<T>::Unknown)?;
+			ensure!(&origin == &d.owner, Error::<T>::NoPermission);
+
+			Metadata::<T>::try_mutate_exists(id, |metadata| {
+				let bytes_used = name.len() + symbol.len();
```

Comment:
note: this is only safe as long as `StringLimit` is sane, can overflow if `StringLimit` is set to `u32::max_value()`

Reply:
Configurations being sane is perfectly valid assumption for pallets.

Not to mention that we would then hit the limit for the tx payload size before we hit this add limit.

Reply:
Let's document what is sane, though?